hathor/simulator/miner/abstract_miner.py

```diff
@@ -0,0 +1,65 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from structlog import get_logger
+from twisted.internet.interfaces import IDelayedCall
+
+from hathor.manager import HathorManager
+from hathor.pubsub import EventArguments, HathorEvents
+from hathor.util import Random
+
+logger = get_logger()
+
+
+class AbstractMiner(ABC):
```

Comment:
As discussed via slack, indeed there's not much to gain here at this point, but new miner classes could benefit from this structure in the future. It could be removed for now, while we don't see any concrete use

Reply:
I don't see any reason to remove it. You have an initial use, and we can undo it in the future if needed.